app/services/bulk_update/recommendations_uploads/validate_trainee.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module BulkUpdate
+  module RecommendationsUploads
+    # finds and validates a trainee scoped within provider, via trn/hesa/provider_trainee_id
+    class ValidateTrainee
+      def initialize(row:, provider:)
+        @row = row
+        @provider = provider
+        @messages = []
+
+        find!
+        validate!
+      end
+
+      def trainee
+        @trainees.first if @trainees.size == 1
+      end
+
+      def valid?
+        @messages.empty?
+      end
+
+      attr_reader :messages
+
+    private
+
+      attr_reader :row, :provider
+
+      def find!
+        trainees_by_trn!
+        trainees_by_hesa_id!
+        trainees_by_provider_trainee_id!
```

Comment:
There are some errors in the error spreadsheet like:

- incompatible TRN & provider_trainee_id
- incompatible hesa_id & provider_trainee_id
- incompatible TRN & hesa_id

i don't think this handles those yet? if a single trainee is found via the TRN, then the validation would pass and assign that trainee to the row.

Reply:
the validations conflict with themselves a bit and I'm hoping to go over them with @johndoates a bit today.

Reply:
is this still outstanding?